gobblin-modules/gobblin-azkaban/src/main/java/gobblin/data/management/conversion/hive/validation/ValidationJob.java

```diff
@@ -153,10 +164,56 @@ public ValidationJob(String jobId, Properties props) throws IOException {
     EventSubmitter.submit(Optional.of(this.eventSubmitter), EventConstants.VALIDATION_SETUP_EVENT);
 
     this.pool = HiveMetastoreClientPool.get(props, Optional.fromNullable(props.getProperty(HiveDatasetFinder.HIVE_METASTORE_URI_KEY)));
+    Preconditions.checkArgument(props.containsKey(VALIDATION_TYPE_KEY), "Missing property " + VALIDATION_TYPE_KEY);
+    this.validationType = ValidationType.valueOf(props.getProperty(VALIDATION_TYPE_KEY));
+    this.ignoreDataPathIdentifierList = COMMA_BASED_SPLITTER.splitToList(props
+        .getProperty(HIVE_VALIDATION_IGNORE_DATA_PATH_IDENTIFIER_KEY,
+            DEFAULT_HIVE_VALIDATION_IGNORE_DATA_PATH_IDENTIFIER));
+    this.throwables = new ArrayList<>();
   }
 
   @Override
   public void run() throws Exception {
+    if (this.validationType == ValidationType.COUNT_VALIDATION) {
+      runCountValidation();
+    } else if (this.validationType == ValidationType.FILE_FORMAT_VALIDATION) {
+      runFileFormatValidation();
+    }
+  }
+
+  /**
+   * Validates that partitions are in a given format
+   */
+  private void runFileFormatValidation()
+      throws IOException {
+    Preconditions.checkArgument(this.props.containsKey(VALIDATION_FILE_FORMAT_KEY));
+    Iterator<HiveDataset> iterator = this.datasetFinder.getDatasetsIterator();
+    EventSubmitter.submit(Optional.of(this.eventSubmitter), EventConstants.VALIDATION_FIND_HIVE_TABLES_EVENT);
+    while (iterator.hasNext()) {
+      HiveDataset hiveDataset = iterator.next();
+      if (!HiveUtils.isPartitioned(hiveDataset.getTable())) {
+        return;
+      }
+      for (Partition partition : hiveDataset.getPartitionsFromDataset()) {
+        if (!shouldValidate(partition)) {
+          return;
+        }
+        String fileFormat = this.props.getProperty(VALIDATION_FILE_FORMAT_KEY);
+        if (!partition.getDataLocation().toString().endsWith(fileFormat)) {
```

Comment:
Couple of observations: 
1. The physical file format comparison should not just limit to comparing file extension but check for magic bytes / file header if possible 
2. We should also validate that the logical SerDe is also as expected by doing a Hive Describe on the partition